Insight 4 - which zipcode has the best rent/price ratio

In [1]:
import os
from spark_utils import *
from pyspark.sql import functions as F

spark = create_spark_session()
bucket = 's3a://helenaudacitybucket'

In [2]:
load_zillow = spark.read.parquet(os.path.join(bucket, 'processed_data', 'Zillow_price_rent'))
load_zillow.createOrReplaceTempView('zillow')

which zipcode has the lowest rent/price ratio

In [3]:
output = spark.sql("""
SELECT State, Metro, CountyName, Zipcode, 2021_07_Rent, 2021_07_Price, (2021_07_Rent/2021_07_Price) AS Rent_Price_Ratio
FROM zillow
ORDER BY Rent_Price_Ratio
LIMIT 10
""")
output.limit(100).toPandas()

State                           Metro          CountyName  Zipcode  \
0    AZ         Phoenix-Mesa-Scottsdale     Maricopa County    85253   
1    CA  Los Angeles-Long Beach-Anaheim  Los Angeles County    90049   
2    TX               Austin-Round Rock       Travis County    78746   
3    CA  Los Angeles-Long Beach-Anaheim  Los Angeles County    90402   
4    CA  San Jose-Sunnyvale-Santa Clara  Santa Clara County    94301   
5    CA  Los Angeles-Long Beach-Anaheim  Los Angeles County    90212   
6    NY     New York-Newark-Jersey City     New York County    10012   
7    CA  Los Angeles-Long Beach-Anaheim       Orange County    92663   
8    CA  San Jose-Sunnyvale-Santa Clara  Santa Clara County    94040   
9    CA  Los Angeles-Long Beach-Anaheim  Los Angeles County    90004   

   2021_07_Rent  2021_07_Price  Rent_Price_Ratio  
0        1775.0      2244662.0          0.000791  
1        2922.0      3118389.0          0.000937  
2        1735.0      1776052.0          0.000977  
3        4478.0      4495269.0          0.000996  
4        4284.0      4145161.0          0.001033  
5        3006.0      2785775.0          0.001079  
6        2706.0      2491808.0          0.001086  
7        2347.0      2145753.0          0.001094  
8        3015.0      2668636.0          0.001130  
9        1813.0      1554043.0          0.001167

*Take home message 1*: If you want to invest in a rental property, avoid the area listed above. For instance 94301 -- Palo Alto and 94040 -- Mountain view

which zipcode has the highest rent/price ratio

In [4]:
output = spark.sql("""
SELECT State, Metro, CountyName, Zipcode, 2021_07_Rent, 2021_07_Price, (2021_07_Rent/2021_07_Price) AS Rent_Price_Ratio
FROM zillow
ORDER BY Rent_Price_Ratio DESC
LIMIT 10
""")
output.limit(100).toPandas()

State                                  Metro           CountyName  Zipcode  \
0    NY            New York-Newark-Jersey City       Suffolk County    11937   
1    MD              Baltimore-Columbia-Towson       Baltimore City    21223   
2    MI                Detroit-Warren-Dearborn         Wayne County    48227   
3    MI                Detroit-Warren-Dearborn         Wayne County    48228   
4    MO                              St. Louis   Saint Louis County    63136   
5    MI                Detroit-Warren-Dearborn         Wayne County    48219   
6    PA         Philadelphia-Camden-Wilmington  Philadelphia County    19132   
7    DE         Philadelphia-Camden-Wilmington    New Castle County    19801   
8    FL  Miami-Fort Lauderdale-West Palm Beach       Broward County    33313   
9    KY            Louisville/Jefferson County     Jefferson County    40211   

   2021_07_Rent  2021_07_Price  Rent_Price_Ratio  
0       33456.0      1752845.0          0.019087  
1        1144.0        60639.0          0.018866  
2         928.0        53734.0          0.017270  
3         848.0        52451.0          0.016167  
4        1022.0        63710.0          0.016041  
5        1005.0        69744.0          0.014410  
6        1030.0        75142.0          0.013707  
7        1464.0       112191.0          0.013049  
8        1625.0       127827.0          0.012712  
9         916.0        73026.0          0.012543

*Take home message 2*: 
It seems that 11937 is a luxury and pricey community. I won't be able to afford that. Most area listed have a very low rent-to-price ratio. However, after I googled them, their high crime rates are high and the school ratings are lower than average. I don't think they are good investments.